crates/accelerate/src/basis/basis_translator/compose_transforms.rs

```diff
@@ -0,0 +1,206 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2024
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use hashbrown::{HashMap, HashSet};
+use pyo3::{exceptions::PyTypeError, prelude::*};
+use qiskit_circuit::circuit_instruction::OperationFromPython;
+use qiskit_circuit::converters::circuit_to_dag;
+use qiskit_circuit::imports::{GATE, PARAMETER_VECTOR, QUANTUM_CIRCUIT, QUANTUM_REGISTER};
+use qiskit_circuit::operations::OperationRef;
+use qiskit_circuit::packed_instruction::PackedInstruction;
+use qiskit_circuit::parameter_table::ParameterUuid;
+use qiskit_circuit::Qubit;
+use qiskit_circuit::{
+    circuit_data::CircuitData,
+    dag_circuit::{DAGCircuit, NodeType},
+    operations::{Operation, Param},
+};
+use smallvec::SmallVec;
+
+// Custom types
+// TODO: Remove these and use the version from `basis_search`
+pub type BasisTransforms = Vec<(String, u32, SmallVec<[Param; 3]>, CircuitRep)>;
+pub type MappedTransforms = HashMap<(String, u32), (SmallVec<[Param; 3]>, DAGCircuit)>;
+// TODO: Remove these and use the version from `EquivalenceLibrary`
+
+/// Representation of QuantumCircuit which the original circuit object + an
+/// instance of `CircuitData`.
+#[derive(Debug, Clone)]
+pub struct CircuitRep(pub CircuitData);
+
+impl FromPyObject<'_> for CircuitRep {
+    fn extract_bound(ob: &Bound<'_, PyAny>) -> PyResult<Self> {
+        if ob.is_instance(QUANTUM_CIRCUIT.get_bound(ob.py()))? {
+            let data: Bound<PyAny> = ob.getattr("_data")?;
+            let data_downcast: Bound<CircuitData> = data.downcast_into()?;
+            let data_extract: CircuitData = data_downcast.extract()?;
+            Ok(Self(data_extract))
+        } else {
+            Err(PyTypeError::new_err(
+                "Provided object was not an instance of QuantumCircuit",
+            ))
+        }
+    }
+}
+
+impl IntoPy<PyObject> for CircuitRep {
+    fn into_py(self, py: Python<'_>) -> PyObject {
+        QUANTUM_CIRCUIT
+            .get_bound(py)
+            .call_method1("_from_circuit_data", (self.0,))
+            .unwrap()
+            .unbind()
+    }
+}
+
+impl ToPyObject for CircuitRep {
+    fn to_object(&self, py: Python<'_>) -> PyObject {
+        self.clone().into_py(py)
+    }
+}
```

Comment:
Are these used?

Reply:
I don't think so, at least not here. This is leftover code brought over from #12585.